### PR TITLE
Fetch information from Rebrickable API

### PR DESCRIPTION
**Pull request by AnkePfei2fer**
- add search component to storybook
- fetch set information from API and store in server
- fetch stored API information from server via Frontend
- display search results as card
- display message when no results were found

image

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -26,6 +31,28 @@ app.get("/api/users", async (_request, response) => {
   response.send(allUsers);
 });
 
+// Send request to Rebrickable API with set number specified by client
+app.get("/api/sets/search_by_set_number/:set_num", async (req, res) => {
+  const response = await fetch(
+    `https://rebrickable.com/api/v3/lego/sets/${req.params.set_num}/?key=${process.env.API_KEY}`
+  );
+  console.log(response);
```
Comment (mpagels) [inline server/index.ts]: `console.log` here

File: server/index.ts [hunk 2]
```diff
@@ -26,6 +31,28 @@ app.get("/api/users", async (_request, response) => {
...
+  const data = await response.json();
+  console.log(data);
```
Comment (mpagels) [inline server/index.ts]: `console.log` here

File: server/index.ts [hunk 3]
```diff
@@ -26,6 +31,28 @@ app.get("/api/users", async (_request, response) => {
   response.send(allUsers);
 });
 
+// Send request to Rebrickable API with set number specified by client
+app.get("/api/sets/search_by_set_number/:set_num", async (req, res) => {
+  const response = await fetch(
+    `https://rebrickable.com/api/v3/lego/sets/${req.params.set_num}/?key=${process.env.API_KEY}`
```
Comment (mpagels) [inline server/index.ts]: Would recommend saving the information in `process.env.API_KEY`  in a variable at the beginning of the file and just use the variable here.

```js
// at line 7
dotenv.config();
const API_KEY = process.env.API_KEY

// later
`https://rebrickable.com/api/v3/lego/sets/${req.params.set_num}/?key=${API_KEY}`
```

Comment (mpagels) [inline server/index.ts]: The `req.params.set_num` is very easy to read over.
Would recommend saving it in a variable earlier for just better readability
```js
const { set_num } = req.params
const response = await fetch(
    `https://rebrickable.com/api/v3/lego/sets/${set_num}/?key=${process.env.API_KEY}`
```

File: server/index.ts [hunk 4]
```diff
@@ -26,6 +31,28 @@ app.get("/api/users", async (_request, response) => {
...
+  console.log(data);
+  res.send(data);
+});
+
+// Send request to Rebrickable API with theme id
+app.get("/api/theme/search_by_theme_id/:theme_id", async (req, res) => {
+  const response = await fetch(
+    `https://rebrickable.com/api/v3/lego/themes/${req.params.theme_id}/?key=${process.env.API_KEY}`
+  );
+  console.log(response);
```
Comment (mpagels) [inline server/index.ts]: `console.log` here

File: src/app/components/Search/search.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    setSearchResult(result);
+    // console.log(searchResult);
+    setSearchResultDetails(result.detail);
+    // console.log(searchResultDetails);
+
+    // fetch theme from API
+    const themeResponse = await fetch(
+      `http://localhost:3001/api/theme/search_by_theme_id/${result.theme_id}`
+    );
+    // console.log(response);
```
Comment (mpagels) [inline src/app/components/Search/search.tsx]: Would recommend removing the comments here

File: src/app/components/Search/search.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    const themeResult = await themeResponse.json();
+    setThemeSearchResult(themeResult);
+    console.log(themeSearchResult);
```
Comment (mpagels) [inline src/app/components/Search/search.tsx]: another `console.log` here

File: src/app/components/Search/search.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            onChange={(event) => setQuery(event.target.value)}
+          ></input>
+          <button className={styles.button}>Suche</button>
+        </form>
+        {searchResultDetails === "Not found." &&
+          (content = (
+            <span className={styles.message}>
+              Wir haben Dein Set leider nicht gefunden. Versuchs nochmal!
+            </span>
+          ))}
```
Comment (mpagels) [inline src/app/components/Search/search.tsx]: This looks weird to me. Is this working? Not sure about the scope of `content` variable. It looks like you are saving again something in `content`. Can you elaborate what is happening here?

**Review by mpagels — CHANGES_REQUESTED** [review]
Hey Anke,
very good job here. Found some little improvements and one bigger question that is worth to look at. Pls review.

File: src/app/components/Search/search.tsx (hunk 7 above)
Comment (AnkePfei2fer) [reply src/app/components/Search/search.tsx]: Thanks for reviewing! I also struggled here, but it works fine now. The problem was to display the message only after the search button was clicked and no result was found. With onChange the message was displayed every time a new letter was typed in the text field. 
(The content variable is for conditional rendering of the page content.)

File: server/index.ts (hunk 1 above)
Comment (AnkePfei2fer) [reply server/index.ts]: What do you mean? I intend to remove it later...

**Pull request merged**